internal/controller/apisixroute_controller.go

```diff
@@ -321,6 +358,12 @@ func (r *ApisixRouteReconciler) validateBackends(ctx context.Context, tc *provid
 			}
 		} else {
 			tc.Upstreams[serviceNN] = &au
+			if err = r.processExternalNodes(ctx, tc, au); err != nil {
```

Comment:
The backend uses Kubernetes services for service discovery, which should not allow custom endpoints. I don't think this needs to be handled.

Reply:
removed